source/plugin/plugin-meta.rst

```diff
@@ -0,0 +1,93 @@
+===============
+Plugin Metadata
+===============
+
+Adding plugin metadata helps users to identify your plugin more easily by giving them the ability to check the name,
+version, description, or even the authors of a plugin at runtime. It is (or will) be also used heavily when publishing
+plugins on a plugin portal like Ore_.
+
+Currently, Sponge supports the following types of plugin metadata:
+
+* Plugin ID, Plugin Name, Version
+* Description
+* URL (e.g. Website)
+* Authors
+* Plugin dependencies
+
+Plugin annotation
+=================
+
+You can define the additional (optional) plugin metadata on your ``@Plugin`` annotation:
+
+.. code-block:: java
+
+    import org.spongepowered.api.plugin.Plugin;
+
+    @Plugin(id = "myplugin", name = "My Plugin", version = "1.0",
+        description = "This is a very cool plugin I made for me",
+        authors = {"Spongie", "FLARD"},
+        url = "http://example.com")
+
+
+File metadata
+=============
+
+Additionally to plugin metadata defined in the plugin annotation we also recommend plugins to include this metadata in
+an extra file in the JAR, which has several advantages:
+
+* Easier integration for build systems (e.g. contributing the version from the build system into plugin metadata)
+* Easier for plugin portals to obtain the plugin metadata (no parsing of class files necessary)
+
+.. note::
+    **We strongly recommend public plugins to include file metadata.** Plugin portals such as Ore_ will likely require
+    file metadata in the future. See `Using the Annotation Processor`_ for a simple way to generate it. The
+    implementation will print a warning if a plugin is missing file metadata.
```

Comment:
"_may_ print a warning", unless of course the API says it must but I don't think it does